test/e2e/disk_encryption.go

```diff
@@ -12,19 +12,49 @@ import (
 	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	mgmtredhatopenshift20220401 "github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2022-04-01/redhatopenshift"
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 )
 
-var _ = Describe("Encryption at host should be enabled", func() {
-	Specify("each VM should have encryption at host enabled", func() {
+// E2E cluster comes with encryption at host and encryption at rest enabled,
+// and it is a part of E2E set up.
+// However but it is not a standard configuration: by default encryption
+// at host is disabled and encryption at rest is enabled, but with
+// Azure managed encryption key.
+//
+// Ideally we should test permutations of cluster features. For example:
+//   - Both encryption at host and encryption at rest enabled
+//   - Both disabled
+//   - Encryption at host enabled, but encryption at rest disabled
+//   - Encryption at host disabled, but encryption at rest enabled, etc
+// Same for other cluster configurations such as public/private
+// API server visibility. But we are not there yet.
+
+var _ = Describe("Encryption at host", func() {
+	It("must be enabled on the test cluster and each VM must have encryption at host enabled", func() {
 		ctx := context.Background()
 
-		By("getting the resource group where the VM instances live in")
+		By("getting the test cluster resource")
 		oc, err := clients.OpenshiftClustersv20220401.Get(ctx, vnetResourceGroup, clusterName)
 		Expect(err).NotTo(HaveOccurred())
+
+		By("checking that encryption at host is enabled for masters")
+		Expect(oc.OpenShiftClusterProperties).To(Not(BeNil()))
+		Expect(oc.OpenShiftClusterProperties.MasterProfile).To(Not(BeNil()))
+		Expect((*oc.OpenShiftClusterProperties.MasterProfile).EncryptionAtHost).To(Equal(mgmtredhatopenshift20220401.Enabled))
+
+		By("checking that encryption at host is enabled for workers")
+		Expect(oc.OpenShiftClusterProperties).To(Not(BeNil()))
+		Expect(oc.OpenShiftClusterProperties.WorkerProfiles).To(Not(BeNil()))
+		Expect(*oc.OpenShiftClusterProperties.WorkerProfiles).NotTo(BeEmpty())
+		for _, profile := range *oc.OpenShiftClusterProperties.WorkerProfiles {
+			Expect(profile.EncryptionAtHost).To(Equal(mgmtredhatopenshift20220401.Enabled))
+		}
+
+		By("getting the resource group where the VM instances live in")
```

Comment:
I'm not sold on the location of this, it is just above a string manipulation, do we expect this to fail?

Reply:
No, we don't. It is mostly for documentation purposes.

Reply:
I'm not sure I am following you @s-amann. What part don't you like the location of?

Reply:
As I understood it these additional lines are to help with debugging a failing test. This line is describing how we are splitting the resourceID which seems unlikely to fail. I'm not sure it is needed. IMO if we wanted something like this we would Sprintf the resourceGroup into the message after we have pulled it out.